clients/src/test/java/org/apache/kafka/common/utils/ImplicitLinkedHashCollectionTest.java

```diff
@@ -142,8 +143,8 @@ static void expectTraversal(Iterator<TestElement> iterator, Integer... sequence)
             assertEquals(sequence[i].intValue(), element.key, "Iterator value number " + (i + 1) + " was incorrect.");
             i = i + 1;
         }
-        assertTrue(i == sequence.length, "Iterator yieled " + (i + 1) + " elements, but " +
-            sequence.length + " were expected.");
+        assertEquals(i, sequence.length, "Iterator yieled " + (i + 1) + " elements, but " +
```

Comment:
in assertEquals, the first parameter is the one that is expected, hence, in this case it would be `assertEquals(sequence.length, i, ....)`

(same comments for other places)